astroplan/scheduling.py

```diff
@@ -90,6 +90,80 @@ def from_exposures(cls, target, priority, time_per_exposure,
         return ob
 
 
+class Scorer(object):
+    """
+    Returns scores and score arrays from the evaluation of constraints on
+    observing blocks
+    """
+    def __init__(self, blocks, observer, schedule):
+        """
+
+        Parameters
+        ----------
+        blocks : list of `~astroplan.scheduling.ObservingBlock` objects
+            list of blocks that need to be scored
+        observer : `~astroplan.Observer`
+            the observer
+        schedule : `~astroplan.scheduling.Schedule`
+            The schedule inside which the blocks should fit
+        """
+        self.blocks = blocks
+        self.observer = observer
+        self.schedule = schedule
+        # the lists below should change if there are new constraints made
+        self.constraint_list = ['AltitudeConstraint', 'AirmassConstraint', 'AtNightConstraint',
```

Comment:
On this subject, see also http://stackoverflow.com/questions/3862310/how-can-i-find-all-subclasses-of-a-given-class-in-python

```Constraint``` subclasses ```object```, so the approach in this thread will work, i.e you can get a list of all ```Constraints``` using

```python
from astroplan.constraints import Constraint
Constraint.__subclasses__()
```

Note that there's a bug that needs fixing for this to work properly; ```AirmassConstraint``` currently subclasses ```AltitudeConstraint``` rather than ```Constraint```.